stdlib/src/collections/list.mojo

```diff
@@ -184,6 +184,36 @@ struct List[T: CollectionElement](CollectionElement, Sized):
         (self.data + self.size).emplace_value(value^)
         self.size += 1
 
+    @always_inline
+    fn insert(inout self, i: Int, owned value :T):
+        """Insert a value to the list at the given index.
+        `a.insert(len(a), value)` is equivalent to `a.append(value)`.
+
+        Args:
+            i: The index for the value.
+            value: The value to insert.
+        """
+        debug_assert(-self.size <= i <= self.size, "insert index out of range")
+        
+        var normalized_idx = i
+        if i < 0:
+            normalized_idx += len(self)
```

Comment:
Hi, I'm curious. What is the intention of this `normalized_idx`? 

Another, perhaps stupid question (to mojo team) is why we are allowing indices, sizes and capacity to have the `Int` where negative values are not valid?

Reply:
Ah, for indices, is it for allowing syntax like `vec.insert(-2, foo)` where it inserts the value at position `len(vec)-2` ?

Reply:
Oh my bad it should use `normalized_index` instead of `i` in the `for` loop. Thanks for pointing it out.
Yeah It should behave similar to [`a.insert` in python](https://docs.python.org/3/tutorial/datastructures.html)
![image](https://github.com/modularml/mojo/assets/56940034/6373ed16-8f19-41b0-88eb-cf58f0bad05d)

Reply:
Their is some ambiguity in python3 like this:
```python
a = []
a.insert(-2, 11)
print(a) # [11]
a = [1, 2, 3, 4]
a.insert(-10, 4)
print(a) # [4, 1, 2, 3, 4]
```

@Neroro64 , @JoeLoser  how should we handle -ve index in `insert`? python3 allow for something similar to above. 
how to handle the case when `len(a) == 0` and we `a.insert(-1, some_value)` ?

Reply:
Sorry for late response. 
Personally, I don't like negative index because that doesn't really provide any value other than the syntactic sugar. Value types with proper "unit" would be safer and easier to work with, imo. 

But if we want the same behavior as python. I guess something like the following would do:
```python
if (index < 0):
    normalized_index = max(0, len(self) + index)
```

For python `list` any `insert(index, value)` where `index <= -len(self)` is equivalent to `insert(0, value)`

See implementation here: https://github.com/python/cpython/blob/c43f6a4dfaa1c1974141e2f7014a7f98ca3a3f93/Objects/listobject.c#L453

Reply:
This should work. Thanks for the input.